types/types.go

```diff
@@ -671,6 +672,26 @@ type SystemContext struct {
 	CompressionFormat *compression.Algorithm
 	// CompressionLevel specifies what compression level is used
 	CompressionLevel *int
+
+	// testClientRoundTripper is a http.Client.Transport hook for fake responses
+	testClientRoundTripper http.RoundTripper
+}
+
+// RegisterTestClientRoundTripper registers testClientRoundTripper, if set.
+// Calls Transport.RegisterProtocol.
+func (sys *SystemContext) RegisterTestClientRoundTripper(transport *http.Transport, schemes []string) *http.Transport {
```

Comment:
I don’t think the internal unit test implementation should leak to the external API like this.

Instead, unit tests (assuming they don’t span multiple packages) can call the private `newImageSource`/`newImageDestination` with an internal-only extra parameter.